### PR TITLE
Improve finding swap transactions

### PR DESCRIPTION
**Pull request by monokh**
### Description

The method for finding the swap transactions would parse blocks and every transaction in that to find it. This was innefficient and on mainnet could make thousands of network calls (per tx) and take minutes

We need a more efficient way

### Submission Checklist :pencil:

- [x] Add `getAddressTransactions` to BitcoreRPCProvider - One call to get matching transactions in a block

### CODE REVIEW

File: src/providers/bitcoin/BitcoinRPCProvider.js [hunk 1]
```diff
@@ -109,6 +109,20 @@ export default class BitcoinRPCProvider extends JsonRpcProvider {
     return this.jsonrpc('getblockcount')
   }
 
+  async getAddressTransactions (address, start, end) {
+    const transactionIds = []
+    for (const blockNumber of _.range(start, end + 1)) {
+      const block = await this.getMethod('getBlockByNumber')(blockNumber, true)
```
Comment (matthewjablack) [inline src/providers/bitcoin/BitcoinRPCProvider.js:115]: Can't we use `getAddressUtxos` for this? Wouldn't this be easier than checking each block?

File: examples/swap/swap-e2e.html [hunk 2]
```diff
@@ -105,12 +105,12 @@ <h3>Refund swap:</h3>
     /* global $, ChainAbstractionLayer, web3 */
     const { Client, providers, networks, crypto } = ChainAbstractionLayer
     const bitcoin = new Client()
-    bitcoin.addProvider(new providers.bitcoin.BitcoreRPCProvider('https://localhost:18332', 'bitcoin', 'local321'))
+    bitcoin.addProvider(new providers.bitcoin.BitcoreRPCProvider('http://localhost:18332', 'bitcoin', 'local321'))
```
Comment (matthewjablack) [inline examples/swap/swap-e2e.html:108]: switch back to `https`

File: examples/swap/swap-e2e.html [hunk 3]
```diff
@@ -105,12 +105,12 @@ <h3>Refund swap:</h3>
...
     bitcoin.addProvider(new providers.bitcoin.BitcoinLedgerProvider({ network: networks.bitcoin_testnet, segwit: false }))
     bitcoin.addProvider(new providers.bitcoin.BitcoinSwapProvider({ network: networks.bitcoin_testnet }))
 
     const ethereum = new Client()
-    ethereum.addProvider(new providers.ethereum.EthereumRPCProvider('https://localhost:8545'))
+    ethereum.addProvider(new providers.ethereum.EthereumRPCProvider('http://localhost:8545'))
```
Comment (matthewjablack) [inline examples/swap/swap-e2e.html:113]: switch back to `https`

**Review by matthewjablack — APPROVED** [review]
LGTM 🎉 just some small comments

File: examples/swap/swap-e2e.html (hunk 2 above)
Comment (monokh) [reply examples/swap/swap-e2e.html:108]: I actually recall discussing whether https was required and i was under the impression that it was for some reason. I then tried to change to http and it still worked. So due to the default host port of starting a local rpc is http, i thought i could change it to that.

File: src/providers/bitcoin/BitcoinRPCProvider.js (hunk 1 above)
Comment (monokh) [reply src/providers/bitcoin/BitcoinRPCProvider.js:115]: `getAddressUtxos` is a bitcore function. This provider assumes bitcoind. 
we could do with `listunspent` but you need to make sure you are watching the p2sh so i don't see much point in that.

**Pull request merged**